hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/main/java/org/apache/hadoop/yarn/server/router/webapp/FederationInterceptorREST.java

```diff
@@ -1145,10 +1146,66 @@ public SchedulerTypeInfo getSchedulerInfo() {
     throw new NotImplementedException("Code is not implemented");
   }
 
+  /**
+   * This method dumps the scheduler logs for the time got in input, and it is
+   * reachable by using {@link RMWSConsts#SCHEDULER_LOGS}.
+   *
+   * @param time the period of time. It is a FormParam.
+   * @param hsr the servlet request
+   * @return the result of the operation
+   * @throws IOException when it cannot create dump log file
+   */
   @Override
   public String dumpSchedulerLogs(String time, HttpServletRequest hsr)
       throws IOException {
-    throw new NotImplementedException("Code is not implemented");
+
+    try {
+      if (StringUtils.isBlank(time)) {
+        routerMetrics.incrDumpSchedulerLogsFailedRetrieved();
+        throw new IllegalArgumentException("Parameter error, the time is empty or null.");
+      }
+
```

Comment:
Move the try here.

Reply:
Thank you very much for helping to review the code, I will modify the code.

Reply:
You can move the isBlank outside of the try.

Reply:
Thanks for your suggestion, I will modify the code.